### ISSUE TITLE
The plugin "gatsby-plugin-page-creator" created a page with a component path that doesn't match

### ISSUE THREAD

**Issue by pantaley** [gatsbyjs/gatsby#17746]
## Description

On Build or develop in the console I see the following error:

```
warn The plugin "gatsby-plugin-page-creator" created a page with a component path that doesn't match
 the casing of the actual file. This may work locally, but will break on systems which are
case-sensitive, e.g. most CI/CD pipelines.

page.component:     "C:\WORK\BUG\GATSBY-SITE\SRC\PAGES\ТЕСТ.JS/"
path in filesystem: "C:\Work\bug\gatsby-site\src\pages\тест.js"
                         ^^^ ^^^ ^^^^^^ ^^^^ ^^^ ^^^^^ ^^^^ ^^^
```

### Steps to reproduce

Just install the starter (gatsby new gatsby-site) and rename page-2 in Cyrilic (for example: тест):

Here is a repo: https://github.com/pantaley/gatsby

### Environment

 System:
    OS: Windows 10
    CPU: (8) x64 Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
  Binaries:
    npm: 6.4.1 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: 44.18362.329.0
  npmPackages:
    gatsby: ^2.15.18 => 2.15.18
    gatsby-image: ^2.2.19 => 2.2.19
    gatsby-plugin-manifest: ^2.2.17 => 2.2.17
    gatsby-plugin-offline: ^3.0.7 => 3.0.7
    gatsby-plugin-react-helmet: ^3.1.7 => 3.1.7
    gatsby-plugin-sharp: ^2.2.24 => 2.2.24
    gatsby-source-filesystem: ^2.1.24 => 2.1.24
    gatsby-transformer-sharp: ^2.2.15 => 2.2.15

**Comment by jonniebigodes** [comment]
@pantaley i think that your issue is related  gatsby not fully supporting non ascii characters, so you're bound to encounter such errors. Some work is being done, like for instance in discussed in #14126 and in the following pr #14372. But it's still a work in progress. As a workaround can you rename the page

**Comment by pantaley** [comment]
@jonniebigodes to be honest I think it's a dependence issue, because a month ago everything was fine.

**Comment by stephen-zhao** [comment]
I made a comment over at the already closed issue #16721 with a related bug.

I'll include the contents here as well...

Hello.. so it seems I am still getting a very similar issue. I am on Gatsby version 2.15.29 and Gatsby-CLI version 2.7.54, running on WSL Ubuntu.

```
warn The plugin "dev-404-page" created a page with a component path that doesn't match the casing of the actual file.
This may work locally, but will break on systems which are case-sensitive, e.g. most CI/CD pipelines.
page.component:     "/mnt/d/projects/Dev_Projects/stephen-zhao.github.io/.cache/dev-404-page.js"
path in filesystem: "/mnt/d/Projects/Dev_Projects/stephen-zhao.github.io/.cache/dev-404-page.js"
                            ^
```

The same applies to plugin `gatsby-plugin-page-creator` as well as several components inside `node_modules`. Not sure what is going on... Why would gatsby want to change the first uppercase character it encounters to a lowercase (the subsequent uppercase characters are unaffected)?

**Comment by stephen-zhao** [comment]
Nevermind.. looks like a mistake was made on my part.

I had accessed my project directory using the path `/mnt/d/projects/....` rather than `/mnt/d/Projects/....` despite the actual directory being capitalized like the latter.

Since it's a Windows directory, it may be accessed case-insensitively, but in WSL which uses posix paths, running gatsby, it gets a little funky.

Bottom line is check your paths!

**Comment by waltercruz** [comment]
good that it worked @stephen-zhao. I will open a ticket to myself about this (not uppercasing when running on WSL), but I won't have time to implement this week.

**Comment by pantaley** [comment]
The issue still exists in gatsby v2.7.6.

**Comment by LekoArts** [comment]
I also can't reproduce this following your steps. It works just fine in my WSL setup.

**Comment by pantaley** [comment]
@LekoArts I can confirm that with WSL everything is working, but the issue is there with git bash terminal.

**Comment by vladar** [comment]
Should be fixed in gatsby `2.17.17`
